dyno-core/src/main/java/com/netflix/dyno/connectionpool/MultiKeyCompressionOperation.java

```diff
@@ -0,0 +1,18 @@
+package com.netflix.dyno.connectionpool;
+/**
+ * 
+ * Interface to be used for multi key operations, i.e. 
+ * taking as input an vararg like String...
+ *
+ * 
+ * @author ipapapa
+ *
+ * @param <CL>
+ * @param <R>
+ */
+public interface MultiKeyCompressionOperation<CL, R> extends Operation<CL, R> {
+
+	 String[] compressMultiKeyValue(ConnectionContext ctx, String... value);
+
+    String decompressValue(ConnectionContext ctx, String value);
```

Comment:
any reason why this doesnt take a list of strings?

Reply:
In multi-key write operations, we receive `String... keyValues` which is a combination of keys and values and we return either a `String` or ` Long`. In multi-key read operations, the API receives a `String keys` which is only the keys that we want to be read and we return a List of values. The list of values is being transformed through the `CollectionUtils.transform` to each independent components (this is how you had set it up and it looks good to me). Then we take each value and we decompress. Hence, our `compressMultiKeyValue` has numerous inputs and has some logic to perform the compression like selecting every second element which is the values. The `decompressValue` does not have any special logic other than decompressing each value one-by-one